docs/build.md

```diff
@@ -0,0 +1,406 @@
+# Building from the ground up
+
+Going forward, it will be assumed that there are two machines available,
+`node-1` and `node-2` respectively. They can be physical machines, virtual
+machines, or even Docker containers. Let's assume that their IPv4 addressed are:
+
+```sh
+NODE_1_IP=192.168.0.1
+NODE_2_IP=192.168.0.2
+```
+
+Firstly, the dependencies should be installed. Most of them can be installed
+using the package manager of your Linux distribution. In case Erlang 21.x is not
+available through it, please build one using [kerl](https://github.com/kerl/kerl):
+
+```sh
+KERL_CONFIGURE_OPTIONS=
+KERL_CONFIGURE_OPTIONS+="--enable-dirty-schedulers "
```

Comment:
nit: dirty shedulers and kernel poll are enabled by default in OTP 21, IIRC it's not even possible to disable them in that version. also we don't use hipe, so it could be disabled.

Reply:
Are you sure? I have just checked its `configure` scripts, and it seems that both can be disabled and and dirty schedulers are disabled by default. I will remove the HiPe flag since it is unnecessary.

Reply:
http://erlang.org/download/otp_src_21.0.readme

I didn't check that, but according the release notices:

1. It's not possible disable dirty schedulers when building erlang. You can disable them in runtime with `+SDio` 0 and `+SDcpu 0:0`, but io stops working.

```
  OTP-14613    Application(s): erts

               It is not longer possible to disable dirty schedulers
               when building erlang.
```

2. you can disable kernel-polling only in build, now I'm not sure if it's enabled by default.

```
  OTP-14346    Application(s): erts

               The I/O polling functionality of erts has been
               re-written to better make use of the OSs polling
               mechanisms. This change means that erts will now always
               prefer to use a kernel-polling mechanism if possible.
               Also all of the I/O polling has been moved to dedicated
               threads instead of being placed in the scheduler loops.

               As a result of this, the erl options +K and +secio have
               been removed. It is still possible to disable
               kernel-poll, but it has to be done at compile time
               through the configure option --disable-kernel-poll.

               The new erl options +IOt and +IOp can be used to change
               how many IO poll threads and poll sets that erts should
               use. See their respective documentation for more
               details.
```

All of these options work fine with OTP 21, so we can keep all of them in the instructions.